django/core/checks/model_checks.py

```diff
@@ -31,35 +31,131 @@ def check_all_models(app_configs=None, **kwargs):
+            'receiver': description,
+            'signal': signal_name,
+            'model_error': app_model_error(model_key),
+        }
+        return Error(error_msg % params, obj=receiver.__module__, hint=None, id='signals.E001')
+
+    def default_error(model_key, func, args, keywords):
+        error_msg = (
+            "%(op)s contains a lazy reference to %(model)s, "
+            "but %(model_error)s."
+        )
+        params = {
+            'op': func,
+            'model': '.'.join(model_key),
+            'model_error': app_model_error(model_key),
+        }
+        return Error(error_msg % params, obj=func, hint=None)
+
+    # Maps common uses of lazy operations to corresponding error functions
+    # defined above. If a key maps to None, no error will be produced.
+    # default_error() will be used for usages that don't appear in this dict.
+    known_lazy = {
+        ('django.db.models.fields.related', 'resolve_related_class'): field_error,
+        ('django.db.models.fields.related', 'set_managed'): None,
+        ('django.dispatch.dispatcher', 'connect'): signal_connect_error,
+    }
+
+    def build_error(model_key, func, args, keywords):
+        key = (func.__module__, func.__name__)
+        error_fn = known_lazy.get(key, default_error)
+        return error_fn(model_key, func, args, keywords) if error_fn else None
+
+    errors = sorted(filter(None, (
+        build_error(model_key, *extract_operation(func))
+        for model_key in pending_models
+        for func in app_registry._pending_operations[model_key]
+    )), key=lambda error: error.msg)
 
-    errors = []
-    for name in dir(models.signals):
-        obj = getattr(models.signals, name)
-        if isinstance(obj, models.signals.ModelSignal):
-            for reference, receivers in obj.unresolved_references.items():
-                for receiver, _, _ in receivers:
-                    # The receiver is either a function or an instance of class
-                    # defining a `__call__` method.
-                    if isinstance(receiver, types.FunctionType):
-                        description = "The '%s' function" % receiver.__name__
-                    else:
-                        description = "An instance of the '%s' class" % receiver.__class__.__name__
-                    errors.append(
-                        Error(
-                            "%s was connected to the '%s' signal "
-                            "with a lazy reference to the '%s' sender, "
-                            "which has not been installed." % (
-                                description, name, '.'.join(reference)
-                            ),
-                            obj=receiver.__module__,
-                            id='signals.E001'
-                        )
-                    )
     return errors
+
+
+@register(Tags.models)
+def check_lazy_references(app_configs=None, **kwargs):
+    return _check_lazy_references(apps)
```

Comment:
Yep. Especially since almost everywhere that argument is used in Django, it looks like this:

```python
checks.run_checks(app_configs=apps.get_app_configs())
```